src/interactivity/win32/Clipboard.cpp

```diff
@@ -228,48 +231,50 @@ void Clipboard::StoreSelectionToClipboard(const bool copyFormatting)
     const auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
     const auto& buffer = gci.GetActiveOutputBuffer().GetTextBuffer();
     const auto& renderSettings = gci.GetRenderSettings();
+    const auto& fontData = gci.GetActiveOutputBuffer().GetCurrentFont();
+    const auto& fontName = fontData.GetFaceName();
+    const auto fontSizePt = fontData.GetUnscaledSize().height * 72 / ServiceLocator::LocateGlobals().dpi;
+    const auto bgColor = renderSettings.GetAttributeColors({}).second;
+    const auto isIntenseBold = renderSettings.GetRenderMode(::Microsoft::Console::Render::RenderSettings::Mode::IntenseIsBold);
 
     const auto GetAttributeColors = [&](const auto& attr) {
-        return renderSettings.GetAttributeColors(attr);
+        const auto [fg, bg] = renderSettings.GetAttributeColors(attr);
+        // TODO: GetAttributeUnderlineColor calls GetAttributeColors
+        // internally, which is redundant.
+        const auto ul = renderSettings.GetAttributeUnderlineColor(attr);
+        return std::tuple{ fg, bg, ul };
     };
 
-    bool includeCRLF, trimTrailingWhitespace;
+    bool singleLine = false;
     if (WI_IsFlagSet(OneCoreSafeGetKeyState(VK_SHIFT), KEY_PRESSED))
     {
         // When shift is held, put everything in one line
-        includeCRLF = trimTrailingWhitespace = false;
+        singleLine = true;
     }
-    else
+
+    const auto& [selectionStart, selectionEnd] = selection.GetSelectionAnchors(true);
+
+    const auto req = TextBuffer::CopyRequest::FromConfig(buffer, selectionStart, selectionEnd, singleLine, !selection.IsLineSelection(), false);
+    text = buffer.GetPlainText(req);
+    if (copyFormatting)
     {

```

Comment:
can we move the locals for all formatted copy stuff down into this condition? no point in loading up all that stuff in the vast majority of copies!